### ISSUE TITLE
Nullable properties in `TsInterface` annotated classes not exported as optional

### ISSUE THREAD

**Issue by takerukoushirou** [reinforced/Reinforced.Typings#83]
I have the same problem as in issue #49, though with `TsInterface` annotated classes. I'm using .NET Core 2.1 with Reinforced Typings version 1.4.95.

Every nullable property (e.g. `bool?`) will not be exported as optional and I have to explicitly add `[TsProperty(ForceNullable = true)]`.

Reinforced Typings configuration (no fluent configuration):
```cs
[assembly:
    TsGlobal(
        ExportPureTypings = true,
        WriteWarningComment = true,
        CamelCaseForMethods = true,
        CamelCaseForProperties = true,
        UseModules = true,
        DiscardNamespacesWhenUsingModules = true,
        GenerateDocumentation = true
    )]
```
with `RtTargetFile` and `<RtDisable>false</RtDisable>` project properties.

**Comment by chriskooken** [comment]
@takerukoushirou I am having the same issue.

**Comment by pavel-b-novikov** [comment]
Well, actually it is really weird feature request since, AFAIK, every property in TS can be set to null - you do not need something special for that. `?` with field actually means that field *may* be omitted when instantiating value of type.

From documentation:

> These optional properties are popular when creating patterns like “option bags” where you pass an object to a function that only has a couple of properties filled in.

So .NET `Nullable<>` has different semantics since it has value/reference types separation. As result, some people insist that C# nullable properties must *not* be translated into TS optional properties, other people insist that they must.

In conclusion, I have decided to add `AutoOptionalProperties` parameter to `[TsGlobal]`/fluent configuration in order to satisfy both sides. If you want nullables to be translated into TS optionals, just turn `AutoOptionalProperties` on. By default it is off. 

Made in version [1.5](https://github.com/reinforced/Reinforced.Typings/blob/features-1.5/Reinforced.Typings.Tests/SpecificCases/SpecificTestCases.AutoOptional.cs)

**Comment by chriskooken** [comment]
Thanks so much, this solves another problem of mine!
